### PR TITLE
docs: document generateDeepReexports compiler option

### PR DESCRIPTION
**Pull request by pauldraper**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Other information

This would have saved me considerable time finding in the source. #45529

### CODE REVIEW

**Comment by jessicajaniuk** [comment]
@alxhub @JoostK This is a quick review. Can either of you take a quick look?

File: aio/content/guide/angular-compiler-options.md [hunk 1]
```diff
@@ -126,6 +126,12 @@ When `true` (the default), generates factory files (`.ngfactory.js` and `.ngstyl
 
 When `false`, factory files are generated only for `.ts` files. Do this when using factory summaries.
 
+### `generateDeepReexports`
+
+When `true`, adds TypeScript re-exports of NgModules' exported directives/pipes.
+
+This option should be enabled for libraries that do not not follow the [Angular Package Format](guide/angular-package-format), which requires that NgModules and directives/pipes are exported from a package entrypoint.
```
Comment (JoostK) [inline aio/content/guide/angular-compiler-options.md:133]:
```suggestion
This option should be enabled for libraries that do not follow the [Angular Package Format](guide/angular-package-format), which requires that NgModules and directives/pipes are exported from a package entrypoint.
```

This should probably also mention that the flag has no effect if `flatModuleOutFile` is configured, as well as mention that this defaults to `false`.

**Comment by bob-watson** [comment]
@pauldraper, Thanks for the pull request. It looks like it's close and I'm following up to see what you'd like to do. Can you let me know how you'd like to proceed, please? Thanks!

**Comment by jessicajaniuk** [comment]
Closing due to inactivity, but feel free to re-open if you'd like to address the remaining issues and to land this PR.

**Pull request closed without merging**